src/common/libflux/util.h

```diff
@@ -0,0 +1,39 @@
+/************************************************************\
+ * Copyright 2022 Lawrence Livermore National Security, LLC
+ * (c.f. AUTHORS, NOTICE.LLNS, COPYING)
+ *
+ * This file is part of the Flux resource manager framework.
+ * For details, see https://github.com/flux-framework.
+ *
+ * SPDX-License-Identifier: LGPL-3.0
+\************************************************************/
+
+#ifndef _FLUX_CORE_UTIL_H
+#define _FLUX_CORE_UTIL_H
+
+#include <flux/core.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+typedef enum flux_process_scope {
+    FLUX_PROCESS_SCOPE_NONE = 0,
+    FLUX_PROCESS_SCOPE_SYSTEM_INSTANCE = 1,
+    FLUX_PROCESS_SCOPE_USER_INSTANCE_INITIAL_PROGRAM = 2,
+    FLUX_PROCESS_SCOPE_FLUX_JOB_INSTANCE_INITIAL_PROGRAM = 3,
+    FLUX_PROCESS_SCOPE_FLUX_JOB_INSTANCE_JOB = 4,
+    FLUX_PROCESS_SCOPE_UNKNOWN = 99,
+} flux_process_scope_t;
```

Comment:
Not a full review, just wanted to get this comment in for consideration:

I would start with a pared down list of scopes with less egregiously long names.  :wink: I _think_ this is all our use cases need at this point? Callers that need to differentiate can do so via other attributes, etc.

```suggestion
typedef enum flux_process_scope {
    FLUX_PROCESS_SCOPE_NONE = 0,
    FLUX_PROCESS_SCOPE_SYSTEM = 1,
    FLUX_PROCESS_SCOPE_INITIAL_PROGRAM = 2,
    FLUX_PROCESS_SCOPE_JOB = 3,
    FLUX_PROCESS_SCOPE_UNKNOWN = 99,
} flux_process_scope_t;
```

Also, I didn't look at the code yet, is there really a need for `FLUX_PROCESS_SCOPE_UNKNOWN`? Shouldn't the function just return an error if it cannot determine the scope?

Reply:
> Shouldn't the function just return an error if it cannot determine the scope?

Because we don't cover every permutation (per my comment in #3817), I wanted to have some flexibility to allow for future additions, thus the UNKNOWN and why I set that to 99.  We could just return an error instead.  Hmmmm EPERM?  Will have to think about it.

Reply:
Aren't the extra permutations just special cases of the already covered scopes as described above?

Reply:
yeah, that's true.  given the more limited set, the potential for future extensions is less likely.